### DISCUSSION TITLE
Single Logout, SAML signature verification and no assertion

### DISCUSSION THREAD

**Discussion by LevN0** [keycloak/keycloak#31540] [Q&A]
### Description

I am consuming a SAML identity provider that does not include a assertion in response to a logout request. Azure AD appears to be an [example](https://learn.microsoft.com/en-us/answers/questions/1184652/invalid-signature-of-saml-logout-response-in-keycl) of such an identity provider, although it's not the one I am using. Example response,

```xml
<samlp:LogoutResponse xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
                      Destination="https://<example.org>/realms/master/broker/<broker_alias>/endpoint"
                      ID="sed1e027ded784c2ab07eb90bfc9c9b2bd7602075"
                      InResponseTo="ID_50b3faa6-058a-4b0e-9110-d571dd649330"
                      IssueInstant="2024-07-22T23:09:23Z"
                      Version="2.0"
                      >i
    <saml:Issuer xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">issuer_name</saml:Issuer>
    <samlp:Status>
        <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success" />
    </samlp:Status>
</samlp:LogoutResponse>
```

When SAML signature verification is enabled, upon logout I get "Invalid signature in response from identity provider." I think it's coming from [this check](https://github.com/keycloak/keycloak/blob/25.0.2/services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java#L693) in the code.

It's not clear to me whether this is a bug (i.e. should it try to validate a signature if there is no assertion in a logout response?) or a feature request (i.e. add option to ignore signature validation on logout requests).

I am using keycloak 25.0.1, but looking at 25.0.2 it does not appear to have any change for this.

**Comment by pixelhomie** [comment]
I have the same Problem. With Azure AD as IDP there is no signature sent in logout response which leads to "Invalid signature in response from identity provider." and even with the current Version of Keycloak 26 it is not possible to disable Signature Verification only for Logout.

**Comment by cgeorgilakis** [comment]
We have the opposite problem. AuthnRequests no signed and logout request request signed.
In SAML Identity Providers AuthnRequests and logout sometimes does not have the same behaviour about signed requests.

That's why keycloak need to support a new field _Want Logout Requests Signed_ that will control if logout will be signed.
We have develop this feature for our instance. Whenever possible, I could create a PR for fixing it and @LevN0 @pixelhomie you need to support it.
Or you can make your own PR sooner.
